features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/internal/FlagsFeatureTest.kt

```diff
@@ -141,6 +148,132 @@ internal class FlagsFeatureTest {
 
     // endregion
 
+    // region Graceful Mode Policy
+
+    @Test
+    fun `M log through internalLogger W logErrorWithPolicy() { release build }`() {
+        // Given - release build (not debuggable)
+        val releaseAppInfo = android.content.pm.ApplicationInfo().apply { flags = 0 }
+        val releaseContext = org.mockito.kotlin.mock<Context>()
+        whenever(releaseContext.applicationInfo).thenReturn(releaseAppInfo)
+        val config = FlagsConfiguration.Builder().build()
+        testedFeature = FlagsFeature(mockSdkCore, config)
+        testedFeature.onInitialize(releaseContext)
+
+        // When
+        testedFeature.logErrorWithPolicy("test message", InternalLogger.Level.ERROR)
+
+        // Then
+        verify(mockInternalLogger).log(
+            eq(InternalLogger.Level.ERROR),
+            eq(InternalLogger.Target.USER),
+            any(),
+            eq(null),
+            eq(false),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M log through internalLogger W logErrorWithPolicy() { release build, gracefulModeEnabled false }`() {
+        // Given - release build should ignore gracefulModeEnabled setting
+        val releaseAppInfo = android.content.pm.ApplicationInfo().apply { flags = 0 }
```

Comment:
probably apply is redundant, because `ApplicationInfo` is created by default with `flags = 0`